eclipse-collections-code-generator/src/main/resources/impl/stack/mutable/synchronizedPrimitiveStack.stg

```diff
@@ -419,7 +419,10 @@ public class Synchronized<name>Stack
     @Override
     public Mutable<name>Stack asUnmodifiable()
     {
-        return new Unmodifiable<name>Stack(this);
+        synchronized (this.lock)
+        {
+            return new Unmodifiable<name>Stack(this);
+        }
     }
```

Comment:
This fix in `asUnmodifiable()` and the similar fix in `toImmutable()` look good, but these may be better to be made in a separate commit.

I see similar issues still exist in `synchronizedPrimitiveBag.stg`, `synchronizedPrimitiveList.stg` and  `synchronizedPrimitiveSet.stg` as well, so I think they should be fixed altogether in a different commit (not necessarily in this PR).

Reply:
I would suggest a different PR as well. Helps in compiling the release notes and shows up in history for tracking